src/lib/coda_intf/staged_ledger_intf.ml

```diff
@@ -297,61 +299,77 @@ module type Transaction_snark_scan_state_generalized_intf = sig
       -> (unit, Error.t) result M.t
   end
 
-  val empty : unit -> t
+  (*All the transactions with undos*)
+  module Staged_undos : sig
+    type t
+
+    val apply : t -> ledger -> unit Or_error.t
+  end
+
+  module Bundle : sig
+    type 'a t = 'a list
+  end
 
-  val capacity : t -> int
+  val staged_undos : t -> Staged_undos.t
+
+  val empty : unit -> t
 
   val fill_work_and_enqueue_transactions :
        t
     -> Transaction_with_witness.t list
     -> transaction_snark_work list
-    -> (ledger_proof * transaction list) option Or_error.t
+    -> ((ledger_proof * transaction list) option * t) Or_error.t
 
   val latest_ledger_proof :
     t -> (Ledger_proof_with_sok_message.t * transaction list) option
 
   val free_space : t -> int
 
-  val next_k_jobs : t -> k:int -> Available_job.t list Or_error.t
-
-  val next_jobs : t -> Available_job.t list Or_error.t
-
-  val next_jobs_sequence : t -> Available_job.t Sequence.t Or_error.t
-
   val base_jobs_on_latest_tree : t -> Transaction_with_witness.t list
 
-  val is_valid : t -> bool
-
   val hash : t -> staged_ledger_aux_hash
 
-  val staged_transactions : t -> ledger_undo list
-
-  val all_transactions : t -> transaction list Or_error.t
-
-  val extract_from_job :
-       Available_job.t
-    -> ( ledger_undo * transaction_snark_statement * transaction_witness
-       , ledger_proof * ledger_proof )
-       Either.t
-
-  val copy : t -> t
+  (** All the transactions in the order in which they were applied*)
+  val staged_transactions : t -> transaction list Or_error.t
 
+  (** Available space and the corresponding required work-count in one and/or two trees (if the slots to be occupied are in two different trees)*)
   val partition_if_overflowing : t -> Space_partition.t
 
   val statement_of_job : Available_job.t -> transaction_snark_statement option
 
-  val current_job_sequence_number : t -> int
-
   val snark_job_list_json : t -> string
 
-  val all_work_to_do :
-    t -> transaction_snark_statement list Sequence.t Or_error.t
+  (** All the proof bundles *)
+  val all_work_statements : t -> transaction_snark_statement Bundle.t list
+
+  (** Required proof bundles for a certain number of slots *)
+  val required_work_pairs : t -> slots:int -> Available_job.t Bundle.t list
 
-  val current_job_count : t -> int
+  (**K proof bundles*)
+  val k_work_pairs_for_new_diff : t -> k:int -> Available_job.t Bundle.t list
 
-  val work_capacity : int
+  (** All the proof bundles for 2^transaction_capacity_log2 slots that can be used up in one diff *)
```

Comment:
Is `^` not used to denote exponentiation anymore?

Reply:
It can get confused with `xor` or other overloadings of the `^` operator. In general I like when math expressions in comments are expressed as programming language expressions.